article/app/controllers/LiveBlogController.scala

```diff
@@ -80,10 +81,13 @@ class LiveBlogController(
     Action.async { implicit request: Request[AnyContent] =>
       val filter = shouldFilter(filterKeyEvents)
       val range = getRange(lastUpdate)
+
       mapModel(path, range, filter) {
         case (blog: LiveBlogPage, _) if rendered.contains(false) => getJsonForFronts(blog)
-        case (blog: LiveBlogPage, blocks) if request.forceDCR    => Future.successful(renderGuuiJson(blog, blocks, filter))
-        case (blog: LiveBlogPage, _)                             => getJson(blog, range, isLivePage, filter)
+        case (blog: LiveBlogPage, blocks) if request.forceDCR && lastUpdate.isEmpty =>
+          Future.successful(renderGuuiJson(blog, blocks, filter))
+        case (blog: LiveBlogPage, blocks) =>
+          getJson(blog, range, isLivePage, filter, blocks.requestedBodyBlocks.map(_.toMap).getOrElse(Map.empty))
```

Comment:
This confused me a little, Scala was complaining that requestedBodyBlocks wasn't the right kind of Map which is why I'm using toMap here.

Reply:
![image](https://user-images.githubusercontent.com/21217225/154316172-8282f398-a5f0-4e42-b05c-f1714708af45.png)

Reply:
if you compile in the terminal you will sometimes get a better error than the one in Intellij btw, like in this case:

[error]  found   : Map[String,Seq[com.gu.contentapi.client.model.v1.Block]] (in scala.collection) 
[error]  required: Map[String,Seq[com.gu.contentapi.client.model.v1.Block]] (in scala.collection.immutable)